EventGenerator/src/CaloCalibGun_module.cc

```diff
@@ -0,0 +1,284 @@
+      double xLgTor = sign[xsn]*(radLgTor + pipeR*cos(theta))*cos(phiLgTor);
+      double yLgTor = sign[ysn]*(radLgTor + pipeR*cos(theta))*sin(phiLgTor);
+      // circulus (center) of the large torus
+      double circLgTor = radLgTor * phi_lbd[idx]  * piconst / 2. / 180.;
+
+      // The phi range for the small torus
+      double phiSmTor = piconst * (_randFlat.fire() * phi_sbd[idx] + 180. + phi_lbd[idx]/2. - phi_sbd[idx])/180.;
+      // x, y, z position of the small torus
+      double xSmTor = sign[xsn]*((radSmTor + pipeR*cos(theta))*cos(phiSmTor) + xsmall + xdistance * idx);
+      double ySmTor = sign[ysn]*((radSmTor + pipeR*cos(theta))*sin(phiSmTor) + ysmall[idx]);
+      // circulus (center) of the small torus
+      double circSmTor = radSmTor * piconst * phi_sbd[idx] / 180.;
+
+      // strait pipe
+      //double xmanifold = rInnerManifold * cos(piconst * (90 - phi_end[idx])/180.);
+      double ymanifold = rInnerManifold * sin(piconst * (90 - phi_end[idx])/180.);
+      double xstart = xsmall + xdistance * idx - radSmTor * cos(piconst * phi_end[idx]/180.);
+      double ystart = ysmall[idx] + radSmTor * sin(piconst * phi_end[idx]/180.);
+      // height of the strait pipe
+      double hPipe = (ymanifold - ystart) / sin(piconst * (90 - phi_end[idx]) / 180.);
+      // a cylinder along y-axis
+      double y_center = _randFlat.fire() * hPipe;
+      double xPipe = pipeR * cos(theta);
+      double xStrait = sign[xsn] * (xPipe * cos(-piconst * phi_end[idx] / 180.) - y_center * sin(-piconst * phi_end[idx] / 180.) + xstart);
+      double yStrait = sign[ysn] * (xPipe * sin(-piconst * phi_end[idx] / 180.) + y_center * cos(-piconst * phi_end[idx] / 180.) + ystart);
+      double lenStrait = hPipe;
+
+      double sample = _randFlat.fire();
+      if(sample <= circLgTor / (circLgTor + circSmTor + lenStrait))
+      {
+        xpipe = xLgTor;
+        ypipe = yLgTor;
+      }
+      else if(sample > circLgTor / (circLgTor + circSmTor + lenStrait) && sample <= (circLgTor + circSmTor) / (circLgTor + circSmTor + lenStrait))
+      {
+        xpipe = xSmTor;
+        ypipe = ySmTor;
+      }
+      else
+      {
+        xpipe = xStrait;
+        ypipe = yStrait;
+      }
+      CLHEP::Hep3Vector pos(xpipe, ypipe, zpipe);
+      // shift the pipe to the front of the calorimeter disk 0
+      pos +=_zPipeCenter;
+
+      //pick time
+      double time = _tmin + _randFlat.fire() * ( _tmax - _tmin );
+
+      //Pick energy and momentum vector
+      double energy =_energy;
+      CLHEP::Hep3Vector p3 = _randomUnitSphere.fire(_energy);
+
+      //Set Four-momentum
+      CLHEP::HepLorentzVector mom(0,0,0,0);
+      mom.setPx( p3.x() );
+      mom.setPy( p3.y() );
+      mom.setPz( p3.z() );
+      mom.setE( energy );
+
+      // Add the particle to  the list.
+      output->emplace_back(PDGCode::gamma, GenId::CaloCalib, pos, mom, time);
+      event.put(std::move(output));
+      // Fill histograms.
+      if ( _doHistograms)
```

Comment:
I know that many of the old generators have embedded histogram utilities.  Our present preference is to move the histograms to a separate module; please do so.

Reply:
Done - removed